common.sh

```diff
@@ -160,9 +175,247 @@ exit_fixed()
+    # First paramater is configuration directory where we want to store a file.
+    #
+    # second parameter is a file which we store
+    # Usage:    add_config_file <clean,dirty> <full_path_file_name>
+
+    config_type=$1
+    filename=$1
+    [ -z "$1" ] && {
+        log_error "Missing configuration type directory."
+        return 1
+    }
+    [ -z "$2" ] && {
+        log_error "Missing full_path_file name."
+        return 1
+    }
+    CLEAN_FILES=${CONFIG_FILES_DIR}/clean
+    DIRTY_FILES=${CONFIG_FILES_DIR}/dirty
+    UNCHECKED_FILES=${CONFIG_FILES_DIR}/unchecked
+    backup_file=${CONFIG_FILES_DIR}/${config_type}/${filename}
+    case "$config_type" in
+        "clean")
+            _check_config_file "dirty" "$filename" \
+            || log_warning "You want to store the file into ${DIRTY_FILES} directory."
+            _check_config_file "clean" "$filename" && return 1
+            ;;
+        "dirty")
+            _check_config_file "clean" "$filename" \
+            || log_warning "You want to store the file into ${CLEAN_FILES} directory."
+            _check_config_file "dirty" "$filename" && return 1
+            ;;
+        "unchecked")
+            _check_config_file "clean" "$filename" \
+            || log_warning "You want to store the file into ${CLEAN_FILES} directory."
+            _check_config_file "dirty" "$filename" \
+            || log_warning "You want to store the file into ${DIRTY_FILES} directory."
+            _check_config_file "unchecked" "$filename" && return 1
+            ;;
+        "deprecated")
+            _check_config_file "clean" "$filename" \
+            || log_warning "You want to store the file into ${CLEAN_FILES} directory."
+            _check_config_file "dirty" "$filename" \
+            || log_warning "You want to store the file into ${DIRTY_FILES} directory."
+            _check_config_file "unchecked" "$filename" \
+            || log_warning "You want to store the file into ${UNCHECKED_FILES} directory."
+            _check_config_file "deprecated" "$filename" && return 1
+            ;;
+        *)
+            log_error "Unknown '$config_type' option in function add_config_file."
+            log_error "Allowed options are <clean,dirty,unchecked,deprecated>."
+            return 1
+            ;;
+    esac
+    mkdir -p `dirname ${backup_file}`
+    cp -a ${filename} ${backup_file}
+    return 0
+}
+
+add_kickstart_dir()
+{
+    # Add filename /root/preupgrade/kickstart directory
+    # File as first parameter has to exists in modules directory
+    # Usage:    add_to_kickstart_dir <filename> <description>
+    script_name=$1
+    kickstart_name="$KICKSTART_DIR/`basename $script_name`"
+    [ ! -f "$script_name" ] && log_error "$script_name does not exist." && exit_error
+    [ -f "$kickstart_name" ] && log_warning "`basename $script_name` already exists in $KICKSTART_DIR"
+    cp -af ${script_name} ${kickstart_name}
+}
```

Comment:
If it adds files, why it's called `add_kickstart_dir`?  Also, can't we add directories?